substrate/primitives/consensus/beefy/src/lib.rs

```diff
@@ -172,12 +172,15 @@ pub mod ecdsa_bls_crypto {
 		<MsgHash as Hash>::Output: Into<[u8; 32]>,
 	{
 		fn verify(&self, signature: &<Self as RuntimeAppPublic>::Signature, msg: &[u8]) -> bool {
-			// `w3f-bls` library uses IETF hashing standard and as such does not exposes
-			// a choice of hash to field function.
-			// We are directly calling into the library to avoid introducing new host call.
-			// and because BeefyAuthorityId::verify is being called in the runtime so we don't have
-
-			EcdsaBlsPair::verify(signature.as_inner_ref(), msg, self.as_inner_ref())
+			// We can not simply call
+			// `EcdsaBlsPair::verify(signature.as_inner_ref(), msg, self.as_inner_ref())`
+			// because that invokes ecdsa default verification which perfoms blake2 hash
+			// which we don't want.
```

Comment:
to avoid planting calcified assumptions & since what is wanted may change in future, it would help readers without context to concretely explain why this is undesired (something as simple as "but this is incompatible since we use keccak when signing").

Reply:
I assume it was because Keccak ends up being cheaper. I looked around a bit and my understanding is that, only the  compression function of Blake2b is [included](https://github.com/ethereum/EIPs/issues/152) as precompiled and you need to iterate over your data in order to get all of it hashed. [This paper](https://www.scitepress.org/Papers/2021/106066/106066.pdf) claims that blake2b is faster but more expensive: "A user would essentially be charged three times or more for submitting a Blake2 transaction in comparison with a Keccak transaction, despite the effective hash real lower cost in execution. that is does not end up cheap comparing to Keccak."
However, I haven't made this choice nor I'm an expert on Ethereum, I think someone from Snowfork team perhaps is in better position to add the clarification here.